### PR TITLE
Operation CongressVoting

### LINKED ISSUE #311: Congress Voting: Messages and API
Proposal about congress voting and voting result should be stored in the blockchain.
We should implement the transaction operation type about proposal and voting result and define operation data.

### PR DESCRIPTION
**Pull request by kfangw**
### Github Issue
Resolves #311 

### Background
To store proposal/result of congress voting to blockchain, new operations needed

### Solution
New 2 operations

### Possible Drawbacks
Operations are just stored in the blockchain with its transaction but no affect to account.

### CODE REVIEW

**Comment by kfangw** [comment]
To make an integration test case, it is needed to support the client module for API. #471 
This may be revised when I found a problem during the test by the client module.

File: lib/transaction/operation.go [hunk 1]
```diff
@@ -12,8 +12,10 @@ import (
 type OperationType string
 
 const (
-	OperationCreateAccount OperationType = "create-account"
-	OperationPayment                     = "payment"
+	OperationCreateAccount        OperationType = "create-account"
+	OperationPayment                            = "payment"
+	OperationCongressVoting                     = "cv"
+	OperationCongressVotingResult               = "cvr"
```
Comment (spikeekips) [inline lib/transaction/operation.go]: @kfangw How about more expressive name like `congress-voting`, `congress-voting-result`? They needs more spaces, but expressive :)

File: lib/transaction/operation_congress_voting.go [hunk 2]
```diff
@@ -0,0 +1,33 @@
+package transaction
+
+import (
+	"boscoin.io/sebak/lib/error"
+	"encoding/json"
+	"time"
+)
+
+type OperationBodyCongressVoting struct {
+	Contract string
```
Comment (spikeekips) [inline lib/transaction/operation_congress_voting.go]: @kfangw Just suggestion, `Contract []byte`, how about this?

File: lib/transaction/operation_congress_voting.go [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+	Voting   struct {
+		Start string
+		End   string
```
Comment (spikeekips) [inline lib/transaction/operation_congress_voting.go]: @kfangw Starting and ending should be `Block.Height`. :)

File: lib/transaction/operation_congress_voting_result.go [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+
+	var tobeHash = ""
+	for _, u := range o.BallotStamps.Urls {
+		if _, err := url.Parse(u); err != nil {
+			return errors.ErrorInvalidOperation
+		}
+		tobeHash += u
+	}
+
+	hash := common.MakeHash([]byte(tobeHash))
```
Comment (spikeekips) [inline lib/transaction/operation_congress_voting_result.go]: @kfangw When I introduced the first thought of voting result things, the `Hash` is the file checksum of voting result, the file is the downloaded from `Urls`.

File: lib/transaction/operation_congress_voting_result.go [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+	tobeHash = ""
+	for _, u := range o.Voters.Urls {
+		if _, err := url.Parse(u); err != nil {
+			return errors.ErrorInvalidOperation
+		}
+		tobeHash += u
+	}
+
+	hash = common.MakeHash([]byte(tobeHash))
+	if string(hash) != o.Voters.Hash {
```
Comment (spikeekips) [inline lib/transaction/operation_congress_voting_result.go]: @kfangw This is also same with upper comment.

File: lib/transaction/operation_congress_voting_result.go (hunk 4 above)
Comment (kfangw) [reply lib/transaction/operation_congress_voting_result.go]: Then, the `Hash` is not verifiable, right?
@spikeekips

Comment (spikeekips) [reply lib/transaction/operation_congress_voting_result.go]: @kfangw Yeah, I thought so. I can be verifiable by manual.

**Pull request merged**